tests/Unit/ResourceStreamTest.php

```diff
@@ -162,6 +217,16 @@ public function it_knows_the_size(): void
         self::assertSame(mb_strlen($content), $stream->size());
     }
 
+    #[Test]
+    public function it_throws_stream_action_exception_if_fstat_fails(): void
+    {
+        $stream = new ResourceStream(fopen('https://www.google.com/', 'r'));
```

Comment:
this will open up a socket to google (slow tests) - Can these extra added exception cases also be covered with the failing stream instead?

Reply:
OK, using failing stream for this test now.